src/Containers/Form/Form.js

```diff
@@ -30,9 +30,10 @@ export class Form extends Component {
     this.setState({ title })
   }
 
-  setList = (newText) => {
-    let newListItem = { text: newText }
-    let newList = [...this.state.list, newListItem]
+  setList = (newText, index) => {
+    let newListItem = { text: newText, index }
+    let newList = Object.assign([], this.state.list, {[index]: newListItem})
+    
```

Comment:
Ooh I would not have thought to use Object.assign here.

Reply:
just wanted to save a few lines :)